python/google/protobuf/descriptor.py

```diff
@@ -832,8 +832,11 @@ def __init__(self, name, package, options=None, serialized_pb=None,
     super(FileDescriptor, self).__init__(options, 'FileOptions')
 
     if pool is None:
-      from google.protobuf import descriptor_pool
-      pool = descriptor_pool.Default()
+      if api_implementation.Type() == 'cpp':
+        pool = _message.default_pool
```

Comment:
Is this change necessary? This `__init__` is never called in case of C++ descriptors (since `__new__` returned a different object)

Reply:
Yes, the check here and original line 150 should be deleted.